lib/Common/Memory/LargeHeapBlock.cpp

```diff
@@ -484,6 +484,9 @@ LargeHeapBlock::AllocFreeListEntry(size_t size, ObjectInfoBits attributes, Large
 
     header->objectIndex = headerIndex;
     header->objectSize = originalSize;
+#ifdef RECYCLER_WRITE_BARRIER
+    header->hasWriteBarrier = (attributes&WithBarrierBit) == WithBarrierBit;
```

Comment:
Nit: spacing. Also, do we need to do this explicitly? Can't we tell by the allocator instead? Is there a case where objects are allocated with and without write barrier in the same heap block?